dotcom-rendering/src/web/lib/getAdPositions.test.ts

```diff
@@ -0,0 +1,213 @@
+import type { DCRCollectionType } from '../../types/front';
+import { getMobileAdPositions } from './getAdPositions';
+
+const defaultTestCollections: Pick<DCRCollectionType, 'collectionType'>[] = [
+	...Array<number>(12),
+].map(() => ({
+	collectionType: 'fixed/large/slow-XIV',
+}));
+
+describe('Mobile Ads', () => {
+	it(`Should not insert ad after container if it's the first one and it's a thrasher`, () => {
+		const merchHighPosition = 2;
+		const testCollections = [...defaultTestCollections];
+		testCollections.unshift({ collectionType: 'fixed/thrasher' });
+
+		const mobileAdPositions = getMobileAdPositions(
+			testCollections,
+			merchHighPosition,
+		);
+
+		expect(mobileAdPositions).not.toContain(0);
+	});
+
+	// MerchandiseHigh is in position:
+	// 3: when it's a network front and collections are equal or more than 4
+	// 2: when collections are equal or more than 4 and is not a network front
+	// 0: when collections are less than 4
+	it.each([3, 2, 0])(
+		`should not insert ad when merchandise high is in position %i`,
+		(merchHighPosition) => {
+			const mobileAdPositions = getMobileAdPositions(
+				defaultTestCollections,
+				merchHighPosition,
+			);
+			expect(mobileAdPositions).not.toContain(merchHighPosition);
+		},
+	);
+
+	it('Should not insert ad after a thrasher container', () => {
+		const merchHighPosition = 2;
+		const testCollections = [...defaultTestCollections];
+		testCollections.splice(6, 0, { collectionType: 'fixed/thrasher' });
+		testCollections.splice(9, 0, { collectionType: 'fixed/thrasher' });
+
+		const mobileAdPositions = getMobileAdPositions(
+			testCollections,
+			merchHighPosition,
+		);
+
+		expect(mobileAdPositions).not.toContain(7);
+		expect(mobileAdPositions).not.toContain(10);
+	});
+
+	// We used https://www.theguardian.com/uk/commentisfree as a blueprint
+	it('Non-network front, with more than 4 collections, without thrashers', () => {
```

Comment:
@Jakeii Can you please add here the scripts we used to get the list of containers and the list of ad slots 🙏 ?

Reply:
Here's the scripts, not sure of the best place for them to go! https://gist.github.com/Jakeii/7da94271db5335d808da9e6a5817e042

Reply:
We used these two scripts to get the `testCollection` and the expected output in of the tests.

```
// Run this to fetch the collection types from the DCR JSON object
(await fetch('https://www.theguardian.com/uk.json?dcr').then(data => data.json())).pressedPage.collections.map(collection => ({collectionType: collection.collectionType}))
```

```
// Run this in your browser to get the container types for the current frontend rendered front
// First filter takes out the palettes container and the merchandise ones
// Map & 2nd filter gets the container indices under which a mobile ad is inserted.
Array.from(document.querySelectorAll('.fc-container'))
  .filter(
    (container) => !container.matches('#palette-styles-new-do-not-delete') && !container.matches('.fc-container--commercial')
  )
  .map(
    (container, index) => container.nextElementSibling?.matches('.fc-container__mpu--mobile') && index
  )
  .filter((item) => item !== false);
  ```